src/notation/internal/notationparts.cpp

```diff
@@ -571,14 +571,15 @@ void NotationParts::replaceInstrument(const InstrumentKey& instrumentKey, const
     TRACEFUNC;
 
     Part* part = partModifiable(instrumentKey.partId);
+    QString newInstrumentPartName = notation::formatInstrumentTitle(newInstrument.trackName(), newInstrument.trait());
```

Comment:
1. Please, move this line closer to 
score()->undo(new Ms::ChangePart(...))

2. You can remove notation::, because NotationParts is already in that namespace